src/service/ratelimit.go

```diff
@@ -151,6 +153,12 @@ func (this *service) ShouldRateLimit(
 				this.stats.ShouldRateLimit.ServiceError.Inc()
 				finalError = t
 			}
+		case memcached.MemcacheError:
+			{
+				this.stats.ShouldRateLimit.MemcachedError.Inc()
+				finalError = t
+			}
+
```

Comment:
I'm not crazy about this fact that we are leaking the cache implementation abstraction to this layer. Can you potentially refactor this so redis/memcache/etc. errors are handled internal to the implementation or there is some other way of hiding which counter needs incrementing?